src/renderer/main.js

```diff
@@ -27,26 +27,40 @@ Vue.component('FontAwesomeIcon', FontAwesomeIcon)
 Vue.use(VueI18n)
 
 // List of locales approved for use
-const activeLocales = ['en-US', 'en_GB', 'ar', 'bg', 'cs', 'da', 'de-DE', 'el', 'es', 'es-MX', 'fi', 'fr-FR', 'gl', 'he', 'hu', 'hr', 'id', 'is', 'it', 'ja', 'nb_NO', 'nl', 'nn', 'pl', 'pt', 'pt-BR', 'pt-PT', 'ru', 'sk', 'sl', 'sv', 'tr', 'uk', 'vi', 'zh-CN', 'zh-TW']
+const activeLocales = ['system', 'en-US', 'en_GB', 'ar', 'bg', 'cs', 'da', 'de-DE', 'el', 'es', 'es-MX', 'fi', 'fr-FR', 'gl', 'he', 'hu', 'hr', 'id', 'is', 'it', 'ja', 'nb_NO', 'nl', 'nn', 'pl', 'pt', 'pt-BR', 'pt-PT', 'ru', 'sk', 'sl', 'sv', 'tr', 'uk', 'vi', 'zh-CN', 'zh-TW']
 const messages = {}
 /* eslint-disable-next-line */
 const fileLocation = isDev ? 'static/locales/' : `${__dirname}/static/locales/`
 
 // Take active locales and load respective YAML file
 activeLocales.forEach((locale) => {
+  // Import elctrons app object to access getLocale function
+  const { app } = require('@electron/remote')
   try {
     // File location when running in dev
-    const doc = yaml.load(fs.readFileSync(`${fileLocation}${locale}.yaml`))
-    messages[locale] = doc
+    if (locale === 'system') {
+      const systemsLocale = app.getLocale()
+      if (activeLocales.indexOf(systemsLocale) !== -1) {
+        const doc = yaml.load(fs.readFileSync(`${fileLocation}${systemsLocale}.yaml`))
+        messages[locale] = doc
+      } else {
+        const regex = new RegExp(`${systemsLocale}-\\w+`, 'g')
+        const doc = yaml.load(fs.readFileSync(`${fileLocation}${activeLocales.join(' ').match(regex)[0]}.yaml`))
+        messages[locale] = doc
```

Comment:
Ok, I'm gonna ask you to change this entirely, but not because this wouldn't work with some adjustments, it's just not as robust.
First, I'll let you know why this is not good enough as it is now:
1. `-` alone is not enough because some locales have `_` instead. Right now, only `nb_NO` exists,  but there might be more added later on, so it needs to be future proof
2. You're assuming that there will always be at least one existing match (no `try/catch` block), not sure if you just forgot
3. Regex totally works for this but it's not necessary, there's a much simpler way

The best way to do this is to use `filter` on the `activeLocales` array and use a `currentValue.startsWith(systemsLocale)` as the condition for it to pass the test in the arrow function.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

Then you can simply check if the new array is empty or not.
If it isn't, then you can load the yaml using the first element of that array.
There won't be a need for a `try/catch` block, because you're already sure the file exists since the array isn't empty :)

Reply:
Additional Request from my side: Please change the string 'Follow System' to something like 'System Language' or similar. 'Following System' seems a bit out of place here and might confuse people translating it over at Weblate

Reply:
Good ideas implemented them in 3ea9f0e and 9a04312 it's amazing how much you can learn through contributing :D
Thank you for the help :+1: